R/module_teal_with_splash.R

```diff
@@ -102,8 +115,44 @@ srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
       raw_data
     }
 
-    res <- srv_teal(id = "teal", modules = modules, raw_data = raw_data, filter = filter)
-    logger::log_trace("srv_teal_with_splash initialized module with data { toString(get_dataname(data))}.")
+    raw_data_checked <- reactive({
+      data <- raw_data()
+      if (inherits(data, "qenv.error")) {
+        #
+        showNotification(sprintf("Error: %s", data$message), type = "error")
+        logger::log_error(data$message)
+        return(NULL)
+      }
+      if (!inherits(data, "teal_data")) {
+        msg <- "Error: server must return 'teal_data' object"
+        showNotification(msg, type = "error")
+        logger::log_error(msg)
+        return(NULL)
+      }
+
+      is_modules_ok <- check_modules_datanames(modules, teal.data::datanames(data))
+      is_filter_ok <- check_filter_datanames(filter, teal.data::datanames(data))
+
+      if (!isTRUE(is_modules_ok)) {
+        showNotification(is_modules_ok, type = "error")
```

Comment:
Result of `check_modules_datanames` and `check_modules_datanames` may also be too long for the notification box.